### PR TITLE
Have SubProject return a list of directories

### PR DESCRIPTION
**Pull request by mgeplf**
* SUBPROJECT_PATHS is a list of directories of the subprojects,
  it is percolated up such that it can be used for exclusion in
  coverage, for example

### CODE REVIEW

File: SubProject.cmake [hunk 1]
```diff
@@ -225,8 +229,13 @@ if(EXISTS "${CMAKE_CURRENT_SOURCE_DIR}/.gitsubprojects")
         "else()\n"
         "  file(APPEND .gitsubprojects \"git_subproject(${__subproject})\n\")\n"
         "endif()\n")
+     set(SUBPROJECT_PATHS "${SUBPROJECT_PATHS} ${__subproject_dir}" 
+       CACHE INTERNAL SUBPROJECT_PATHS)
```
Comment (mgeplf) [inline SubProject.cmake]: Not a huge fan of namespace pollution: if there's a more idiomatic way of doing this, let me know, and I'll change it

Comment (eile) [inline SubProject.cmake]: A slight better way is to set a global property, at least it's not kept between cmake runs:
```
    set_property(GLOBAL APPEND PROPERTY
                 ${PROJECT_NAME}_ALL_DEP_TARGETS ${LibName})
```

Comment (mgeplf) [inline SubProject.cmake]: Ahh, interesting, I will fix it Monday.  Thanks for the tip, and the patience.

File: SubProject.cmake [hunk 2]
```diff
@@ -225,8 +229,13 @@ if(EXISTS "${CMAKE_CURRENT_SOURCE_DIR}/.gitsubprojects")
         "else()\n"
         "  file(APPEND .gitsubprojects \"git_subproject(${__subproject})\n\")\n"
         "endif()\n")
+      set(__subproject_paths "${__subproject_paths} ${__subproject_dir}")
```
Comment (eile) [inline SubProject.cmake]: ```set_property(GLOBAL APPEND ...)``` instead of the two added lines above?

Comment (mgeplf) [inline SubProject.cmake]: If I remember properly, I think I tried that, but I wanted to de-duplicate the list, so I have to change it into a variable at some point and thus it ended up being the same, just slightly re-arranged.

Comment (eile) [inline SubProject.cmake]: Ah :)

Ok, +1 from me then.

Comment (rdumusc) [inline SubProject.cmake]: replace
set(__subproject_paths "${__subproject_paths} ${__subproject_dir}")
with
list(APPEND __subproject_paths ${__subproject_dir})
and remove separate_arguments(__subproject_paths)?

**Comment by rdumusc** [comment]
ping

**Comment by mgeplf** [comment]
Sorry, missed that.   I'll try and find time today to try that and test it.  Seems reasonable.

**Comment by mgeplf** [comment]
@rdumusc 

Made the changes, thanks for the tip.

**Pull request merged**